packages/plugin-assets-retry/src/runtime/initialChunkRetry.ts

```diff
@@ -38,15 +38,19 @@ function findNextDomain(url: string, domainList: string[]) {
   return domainList[(index + 1) % domainList.length] || url;
 }
 
+function removeQuery(url: string) {
+  return url.replace(/\?.*$/, '');
+}
+
 function getRequestUrl(element: HTMLElement) {
   if (
     element instanceof HTMLScriptElement ||
     element instanceof HTMLImageElement
   ) {
-    return element.src;
+    return removeQuery(element.src);
```

Comment:
It would be better if we can preserve the existed query in the URL (maybe the query is needed by users).

Reply:
有点子难搞，如果用户的 query 是 `?retry=1`，请求失败后，在我们的角度是不清楚 `?retry=1&retry=2` 还是 `?retry=2`

我把这个配置暂时做成会覆盖 query，不支持保留 query

Reply:
Can we just override `retry` and preserve other querys?

Reply:
we export a function type to let user define it just like

```js
addQuery: (times) => {
    return times === 3 ? `?retryCount=${times}&isLast=1` : `?retryCount=${times}`
},
```

Reply:
So how can users preserve the existed queries via the addQuery function?

Reply:
> Can we just override retry and preserve other querys?

Because it is customized by function type, so we can't do this.

> So how can users preserve the existed queries via the addQuery function?

But we can do this. How about we change this function type to `(options: AssetsRetryHookContext) => string`, In this way, users can freely choose how to edit the query.

Reply:
A corner case is url can be set `?a=1&b=2#title`, anchor url cannot be handled